File: LearnOpencv/5/python/main.py
import cv2
import numpy as np

# 读取彩色图像
image = cv2.imread("C:/github/lib/LearnOpencv/1/c++/cqss.png")
if image is None:
    print("Error: Could not open or find the image!")
    exit()

#调整大小便于显示    
image=cv2.resize(image,(320,320),interpolation=cv2.INTER_AREA)

# 通道分离
B,G,R=cv2.split(image)

# 显示图像
cv2.imshow("R",R)
cv2.imshow("G",G)
cv2.imshow("B",B)
cv2.imshow("img",image)

# 等待按键关闭
cv2.waitKey(0)

File: LearnOpencv/5/c++/main.cpp
#include <iostream>
#include <opencv2\highgui\highgui.hpp>
#include <opencv2\opencv.hpp>
using namespace std;
using namespace cv;

int main()
{
    //读取图像
    Mat img=imread("C:/github/lib/LearnOpencv/1/c++/cqss.png");
    //调整图像大小便于显示
    Size newsize(320,320);
    resize(img,img,newsize);
    //建立新数组用来存放分离后的通道
    vector<Mat> c;
    //通道分离
    split(img,c);
    //显示图像
    imshow("B",c[0]);
    imshow("G",c[1]);
    imshow("R",c[2]);
    imshow("img",img);
    //等待按键关闭
    waitKey(0);
    return 0;
}